Tests/Microsoft.AppCenter.Test.Windows/Storage/FakeStorageTest.cs

```diff
@@ -85,11 +85,11 @@ public void GetLogsQueryError()
             var mockAdapter = new Mock<IStorageAdapter>();
             mockAdapter.Setup(
                     a => a.GetAsync(It.IsAny<PredType>(), It.IsAny<int>()))
-                .Throws(new StorageException());
+                .ThrowsAsync(new StorageException());
```

Comment:
You can use
```C#
.Throws(TaskExtension.GetFaultedTask(new StorageException()))
```
to avoid moq bug.

Reply:
This line of code produce an error `Error CS1503	Argument 1: cannot convert from 'System.Threading.Tasks.Task<Microsoft.AppCenter.Storage.StorageException>' to 'System.Exception'`.

Anyway we can't use `Throws()` to mock an exception in async method.
I think that there is an another way to rewrite it. We could use `Returns()` with a faulted task. It will look bulky but will do what was intended.
`.Returns(TaskExtension.GetFaultedTask<List<Microsoft.AppCenter.Storage.Storage.LogEntry>>(new StorageException()))`

I can try to change it in a such way if we don't want to wait for a moq4 release.

Reply:
Oops, I mean `Returns` the faulted task of course. You understood my idea correctly :+1: